### PR TITLE
update kots helm to be able to download from local registry

### PR DESCRIPTION
**Pull request by alvarorm22**
**Description of the change**
- [x] Enable local registry to be able to pull images in airgap mode
- [x] Add `tenantCheckerRequirements` image to airgap bundle

**Benefits**

- [x] Be able to use airgap mode

### CODE REVIEW

**Comment by alvarorm22** [comment]
https://app.shortcut.com/cartoteam/story/416344/update-kots-config-with-local-registry

File: manifests/kots-helm.yaml [hunk 1]
```diff
@@ -561,3 +569,15 @@ spec:
...
+    - when: '{{repl and (IsAirgap) (HasLocalRegistry) }}'
+      recursiveMerge: true
+      values:
+        global:
+          imageRegistry: '{{repl LocalRegistryAddress }}'
+          imagePullSecrets:
+            - '{{repl ImagePullSecretName }}'
+        replicated:
+          imagePullSecrets:
+            - '{{repl ImagePullSecretName }}'
```
Comment (moicalcob) [inline manifests/kots-helm.yaml:583]: We shared some feedback with Replicated team as IMHO the replicated helm chart should be able to reuse the global image pull secrets https://cartoteam.slack.com/archives/C05V5V84W7M/p1718285356392639?thread_ts=1718211262.868209&cid=C05V5V84W7M

File: manifests/kots-helm.yaml [hunk 2]
```diff
@@ -8,10 +8,18 @@ spec:
...
+      enabled: true
+    internalPostgresql:
+      enabled: false
+
   values:
     replicated:
       enabled: true
       platformDistribution: '{{repl ConfigOption "platformDistribution" }}'
+      images:
+        replicated-sdk: '{{repl LocalImageName "docker.io/replicated/replicated-sdk:v1.0.0-beta.21" }}'
```
Comment (moicalcob) [inline manifests/kots-helm.yaml:22]: The SDK does not allow to include just the repo URL so we'll have to maintain this 😮‍💨

File: manifests/kots-app.yaml [hunk 3]
```diff
@@ -23,6 +23,8 @@ spec:
     - deployment/carto-workspace-subscriber
     - deployment/carto-workspace-www
   minKotsVersion: 1.107.4
+  additionalImages:
+    - gcr.io/carto-onprem-artifacts/tenant-requirements-checker:2024.4.17-rc.7
```
Comment (moicalcob) [inline manifests/kots-app.yaml:27]: I've already asked Replicated team why the bundle is not automatically adding this image in it: https://cartoteam.slack.com/archives/C05V5V84W7M/p1718285246347899 For the moment we need this for Verizon

File: manifests/kots-helm.yaml [hunk 4]
```diff
@@ -8,10 +8,18 @@ spec:
   chart:
     name: carto
     chartVersion: 1.110.1
+  builder:
```
Comment (ManuelLR) [inline manifests/kots-helm.yaml:11]: ohh crap, I was saying WTF is doing CARTO builder here? :see_no_evil:

**Review by ManuelLR — APPROVED** [review]

File: manifests/kots-helm.yaml (hunk 4 above)
Comment (moicalcob) [reply manifests/kots-helm.yaml:11]: 😂

**Pull request merged**